cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -35,6 +34,101 @@ namespace lists {
 namespace detail {
 namespace {
 using offset_type = lists_column_view::offset_type;
+
+/**
+ * @brief Performs an equality comparison between two entries in a lists column
+ *
+ * For the two elements that are in the same list in the lists column, they will always be
+ * considered as different. If they are from the same list and their type is one of floating
+ * point types, this functor will return the same comparison result as
+ * `cudf::element_equality_comparator`. For floating point types, entries holding NaN value will
+ * be considered as different.
+ *
+ * @tparam has_nulls Indicates the potential for null values in the column
+ * @tparam Type      The data type of entries
+ */
+template <bool has_nulls, class Type>
```

Comment:
Does the `has_nulls` _have to_ be a template parameter?
Making it a runtime parameter does not usually reduce performance but does have a big impact on compile time.

Reply:
It does not **have** to. I adopted it from the existing code in cudf. Yes, it may be beneficial to change to a runtime parameter.

Reply:
Hah, thanks so much for this. Changing `has_nulls` to runtime parameter reduces compile time by half (1m20s to 43s), object file size reduces from 8.4MB to 5.3MB.